django/cantusdb_project/main_app/permissions.py

```diff
@@ -0,0 +1,140 @@
+from django.db.models import Q
+
+
+def user_can_edit_chants_in_source(user, source):
```

Comment:
Can we annotate all of these functions indicating they return a boolean? `def user_can_edit_chants_in_source(user, source) -> bool:`, and similarly for all the rest?

Reply:
Yes, will do.